LibreNMS/OS/SocomecUps.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/*
+ * SocomecUps.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       https://www.librenms.org
+ * @copyright  2024 Peca Nesovanovic
+ * @author     peca.nesovanovic@sattrakt.com
+ */
+
+namespace LibreNMS\OS;
+
+use App\Models\Device;
+use LibreNMS\OS;
+
+class SocomecUps extends OS
+{
+    public function discoverOS(Device $device): void
+    {
+        parent::discoverOS($device); // yaml
+
+        $data = intval($this->getCacheTable('SOCOMECUPS-MIB::upsConfigNomKva')['upsConfigNomKva.0']);
```

Comment:
Don't cache this with a walk, it isn't a table.
Just run
```suggestion
        $data = intval(SnmpQuery::get('SOCOMECUPS-MIB::upsConfigNomKva.0')->value());
```

Reply:
done